### ISSUE TITLE
[Shared Addresses] Being kicked after unselecting shared accounts

### ISSUE THREAD

**Issue by caybro** [status-im/status-desktop#14422]
# Bug Report
Testing and playing around with the Edit Shared Addresses dialog, I discovered this:

![image](https://github.com/status-im/status-desktop/assets/5377645/06c494b1-5848-41e4-adda-8ec4c8c2e9bb)

I unselected my main account which contains the STATCC token (to verify we're getting the red banner on top and the updated Save button). Noticed that `wakuext_checkCommunityChannelPermissions` was fired in the background.

I didn't save the changes here in the dialog, just left it open, and few minutes later, after the check completed:

![image](https://github.com/status-im/status-desktop/assets/5377645/bfa0ab51-8721-471d-8cdb-68ea3ec31a67)

I got kicked

Tried to reproduce later today again with the same result:
![image](https://github.com/status-im/status-desktop/assets/5377645/81c81ce4-2694-4273-b6fb-5772453748d7)

#### Expected behavior
No kick or permissions altered without saving in the dialog

#### Actual behavior
Kicked as a result of unselecting the crucial account that holds the needed token

### Additional Information

- Status desktop version: master
- Operating System: linux

**Comment by jrainville** [comment]
@caybro I can't manage to reproduce. I am on top of 2.28.1 and I tried multiple times.

Some attempts were made using a small dev community. In this one, both wallets are selected on open. I unselect the main one and nothing happens even if I leave it open for a while

Most of my attempts I made on my main account that I put in dev. I open the Edit Shared Address popup on the Status community.
Then nothing is selected in my case, because it's a re-imported account so the DB doesn't remember my past shared addresses.
I tried multiple options, select both accounts. wait for the evaluation to be done and after unselect the main wallet.
I tried just selecting the empty account.
In all cases, I never got kicked out.

I checked the code, and it doesn't send the update to the control node until we click and that code doesn't have the self-kick feature.

So I think either your branch had something that caused it, but considering other people had it too, I think you might have just been an accidental victim of this bug: https://github.com/status-im/status-desktop/issues/14432
TLDR: the control node can send outdated descriptions by accident. So maybe you joined, the control was doing an evaluation from before you were joined, the evaluation completed and sent an outdated description where you weren't a member. Poof, you're "kicked" (the app checks the member list and if we are no longer member, we count that as kicked)

**Comment by jrainville** [comment]
If you try to reproduce again in your morning (before Iuri is awake), or even on another community, it would help confirm or infirm my theory.

**Comment by caybro** [comment]
> If you try to reproduce again in your morning (before Iuri is awake), or even on another community, it would help confirm or infirm my theory.

Cool, I'll try to reproduce again but it wasn't any special branch, just master. But let's see :)

**Comment by caybro** [comment]
I could still reproduce in master; let's keep this open for now and we'll revisit once we release 2.28.1. It might be that master simply doesn't have all the fixes that we have in this stable branch

**Comment by jrainville** [comment]
> I could still reproduce in master; let's keep this open for now and we'll revisit once we release 2.28.1. It might be that master simply doesn't have all the fixes that we have in this stable branch

Ok, I'll switch the milestone to 2.29. I'll try to reproduce on master as well later. I was indeed only trying the release branch.

**Comment by anastasiyaig** [comment]
i tried just now in release branch and cant reproduce as well

**Comment by jrainville** [comment]
I tried latest master. I still can't reproduce. The fix for https://github.com/status-im/status-desktop/issues/14432 still wasn't merged, so maybe you're still just getting bamboozled by it.

We can revisit later.

**Comment by caybro** [comment]
> I tried latest master. I still can't reproduce. The fix for #14432 still wasn't merged, so maybe you're still just getting bamboozled by it.
> 
> We can revisit later.

Yeah maybe; thanks for testing!

**Comment by caybro** [comment]
Just happened that I simply opened the "Edit Shared Addresses" dialog, didn't touch anything (like the address selection), and while the check was still ongoing, I got kicked from the Status CC community in the background:

![Snímek obrazovky z 2024-04-25 12-22-14](https://github.com/status-im/status-desktop/assets/5377645/81b6728c-6d46-4ca4-9fb0-a02d9c4d8892)

**Comment by jrainville** [comment]
I created a branch on status-go that adds logs: `debug/user-kicked-for-no-reason`.

I still can't reproduce, so it would be good to get some info as to where it happens.

@caybro if you could run that commit and give me the logs when the kick happens.

Also, I'm curious if that issue only happens on the Status community. We've seen this situation where randomly, someone would have the NFT not associated correctly to their account, but we don't know why, so maybe that's the case for you? If so, maybe we just need to re-send you the Status-CC NFT?

**Comment by caybro** [comment]
> I created a branch on status-go that adds logs: `debug/user-kicked-for-no-reason`.
> 
> I still can't reproduce, so it would be good to get some info as to where it happens.
> 
> @caybro if you could run that commit and give me the logs when the kick happens.
> 
> Also, I'm curious if that issue only happens on the Status community. We've seen this situation where randomly, someone would have the NFT not associated correctly to their account, but we don't know why, so maybe that's the case for you? If so, maybe we just need to re-send you the Status-CC NFT?

I'll give it a try once I get accepted back :/

![image](https://github.com/status-im/status-desktop/assets/5377645/3f505bdc-22a5-442c-87e7-7b66122d1b94)

**Comment by jrainville** [comment]
i thought Iuri kept the control node active now :thinking: 
Maybe it's off while he's on vacation. I'll check with him

**Comment by caybro** [comment]
> @caybro were you added back to the Status community? Is it still reproducible on the latest master for you?

Yeah I can pretty much rejoin freely, but I just experienced it yesterday again. Not sure about the cause, I just restarted the app and I was kicked

**Comment by jrainville** [comment]
@MishkaRogachev made a discovery today. It seems that when you remove a permissioned channel, members get kicked, so I wonder if it's related. See point 3 here https://github.com/status-im/status-desktop/issues/14915#issuecomment-2136168365

**Comment by jrainville** [comment]
I'll close this issue as it's superseded by https://github.com/status-im/status-desktop/issues/15009
